CLOUDFRONT.md

```diff
@@ -58,51 +56,60 @@ php admin/cli/scheduled_task.php --execute='\tool_objectfs\task\generate_status_
-
-### Generate keys
-1. Make a directory _$CFG->dataroot . '/objectfs/'_.
-2. Make it readable and writable:
+### Generate CloudFront keys
+1. Generate an RSA key pair with a length of 2048 bits:
 ```
-chmod 777 objectfs
-```
-3. Generate an RSA key pair with a length of 2048 bits:
-```
-cd objectfs/
 openssl genrsa -out cloudfront.pem 2048
-chmod 777 cloudfront.pem
 ```
-4. Extract the public key:
+2. Extract the public key:
 ```
 openssl rsa -pubout -in cloudfront.pem -out public_key.pem
 ```
-5. Navigate to [https://console.aws.amazon.com/cloudfront/v3/home#/distributions].
-6. In the navigation menu, choose _Public keys_.
-7. Click _Create public key_.
-8. Enter key name.
-9. Enter key value. Use the following command to get the public key:
+3. Navigate to https://console.aws.amazon.com/cloudfront/v3/home#/publickey.
+4. Click _Create public key_.
+5. Enter key name.
+6. Enter key value. Use the following command to get the public key:
 ```
 cat public_key.pem
 ```
-10. Click _Create public key_.
-11. Write down key ID from the [https://console.aws.amazon.com/cloudfront/v3/home#/publickey] page.
+7. Click _Create public key_.
+8. Write down key ID from the https://console.aws.amazon.com/cloudfront/v3/home#/publickey page.
+9. Navigate to https://console.aws.amazon.com/cloudfront/v3/home#/keygrouplist
+10. Create a key group and select the public key created previously
+11. Store the public and private key files somewhere secure
+
+### Create CloudFront response headers policy
+
+1. Navigate to https://console.aws.amazon.com/cloudfront/v3/home#/policies/responseHeaders
+2. Click on _Create response headers policy_
+3. _Name_: CORS-with-preflight-and-SecurityHeadersPolicy-ReadOnly
+4. _Configure CORS_: disabled
+5. _Strict-Transport-Security_: Enabled, origin override enabled
+6. _X-Content-Type-Options_: Enabled, origin override enabled
+7. _X-Frame-Options_: Enabled, SAMEORIGIN, origin override enabled
+8. _X-XSS-Protection_: Enabled, block, origin override enabled
+9. _Referrer-Policy_: Enabled, strict-origin-when-cross-origin, origin override enabled
+10. _Content-Security-Policy_: disabled
+
+### Create CloudFront distribution
+1. Navigate to https://console.aws.amazon.com/cloudfront/.
+2. Click on _Create a CloudFront distribution_.
+3. Choose your Amazon S3 bucket from _Origin domain_ dropdown menu.
+4. _S3 bucket access_: Choose _Yes use OAI (bucket can restrict access to only CloudFront)_ and click _Create new OAI_.
+5. _S3 bucket access -> Bucket policy_: Choose _Yes, update the bucket policy_.
+6. _Viewer protocol policy_: Choose _Redirect HTTP to HTTPS_.
+7. _Allowed HTTP methods_: Choose _GET, HEAD, OPTIONS_ and tick _OPTIONS_ under _Cache HTTP methods_.
+8. _Restrict viewer access_: Choose _Yes -> Trusted key groups (recommended)_.
+9. _Cache key and origin requests_: Choose _Cache policy and origin request policy (recommended)_.
```

Comment:
We'd need to add the key groups and choose the key group created above.

Reply:
Added addition text with this step.